server/keyshare/keyshareserver/server.go

```diff
@@ -27,68 +23,60 @@ import (
 	"github.com/go-chi/chi"
 )
 
-type SessionData struct {
-	LastKeyID    irma.PublicKeyIdentifier // last used key, used in signing the issuance message
-	LastCommitID uint64
-	expiry       time.Time
-}
-
-// Used to provide context in protocol sessions
-type requestAuthorization struct {
-	user                  KeyshareUser
-	hasValidAuthorization bool
-}
-
 type Server struct {
 	// configuration
 	conf *Configuration
 
 	// external components
 	core          *keysharecore.Core
 	sessionserver *irmaserver.Server
-	db            KeyshareDB
+	db            DB
 
 	// Scheduler used to clean sessions
 	scheduler     *gocron.Scheduler
 	stopScheduler chan<- bool
 
 	// Session data, keeping track of current keyshare protocol session state for each user
-	sessions    map[string]*SessionData
-	sessionLock sync.Mutex
+	store sessionStore
 }
 
 func New(conf *Configuration) (*Server, error) {
 	var err error
 	s := &Server{
 		conf:      conf,
-		sessions:  map[string]*SessionData{},
+		store:     newMemorySessionStore(10 * time.Second),
 		scheduler: gocron.NewScheduler(),
 	}
 
-	// Do initial processing of configuration and create keyshare core
+	// Setup IRMA session server
+	s.sessionserver, err = irmaserver.New(conf.Configuration)
```

Comment:
Full consistency with the requestor server (constructing the server at the very end within the `return`) is not possible here unfortunately, because some lines below a method is called on `s`. Given that, I don't see much point in moving the construction of `s` downwards here.